Code/GraphMol/Substruct/SubstructUtils.cpp

```diff
@@ -116,6 +135,10 @@ bool atomCompat(const Atom *a1, const Atom *a2,
   } else {
     res = a1->Match(a2);
   }
+  if (res) {
```

Comment:
Maybe avoid making the call if we don't need to?
```suggestion
  if (res && !ps.atomProperties.empty()) {
```

Reply:
Makes sense!